docs/languages/Kotlin.md

```diff
@@ -0,0 +1,47 @@
+# Kotlin
+
+There are special use-cases that each language supports; this document pertains to **Kotlin models**.
+
+Since `data classes` are used for every model that has got properties, there is no need for additional settings or 
+features to generate `toString()`, `equals()`, `hashCode()`,  getters or setters.
+
+Classes without properties are depicted by usual `classes`, they get no `toString()`, `equals()` or `hashCode()`
+implementation. The default one should suffice here.
+
+<!-- toc is generated with GitHub Actions do not remove toc markers -->
+
+<!-- toc -->
+
+- [Include KDoc for properties](#include-kdoc-for-properties)
+- [Change the collection type for arrays](#change-the-collection-type-for-arrays)
+- [Include Javax validation constraint annotations for properties](#include-javax-validation-constraint-annotations-for-properties)
+- [Generate serializer and deserializer functionality](#generate-serializer-and-deserializer-functionality)
+
+<!-- tocstop -->
+
+## Include KDoc for properties
+To generate models containing `KDoc` from description and examples, use the `KOTLIN_DESCRIPTION_PRESET` option.
+
+Check out this [example for a live demonstration](../../examples/kotlin-generate-kdoc).
+
+## Change the collection type for arrays
+
+Sometimes, we might want to render a different collection type, and instead of the default `Array` use as `List` type. To do so, provide the option `collectionType: 'List'`.
+
+Check out this [example for a live demonstration](../../examples/kotlin-change-collection-type).
+
+## Include Javax validation constraint annotations for properties
+
+In some cases, when you generate the models from JSON Schema, you may want to include `javax.validation.constraint` annotations.
+
+Check out this [example for a live demonstration](../../examples/kotlin-generate-javax-constraint-annotation).
+
+## Generate serializer and deserializer functionality
+
+The most widely used usecase for Modelina is to generate models that include serialization and deserialization functionality to convert the models into payload data. This payload data can of course be many kinds, JSON, XML, raw binary, you name it.
+
+As you normally only need one library to do this, we developers can never get enough with creating new stuff, therefore there might be one specific library you need or want to integrate with. Therefore, there is not one specific preset that offers everything. Below is a list of all the supported serialization presets.
+
+As Kotlin integration with the common parsers and mappers, like Jackson, is pretty good, there are no special presets
```

Comment:
I haven't spent much time with Kotlin, so I cant say for sure, so please correct me if I am wrong 😄 

There are two reasons why you would like to have presets for serializing and deserializing the models:
1. Property names from inputs arent guaranteed to align with what a language allows, for example in JSON Schema you could have a property name such as `"SOME %#!"` but in Kotlin you are not able to generate a property with such a name, and instead the constraint functionality create a Kotlin friendly name for it. Here it's important that the property is converted to the `property.unconstrainedPropertyName`, in the same manner we have for Java Jackson preset: https://github.com/asyncapi/modelina/blob/9e3a823dd7ccd5f206f1028277189291a51f56b3/src/generators/java/presets/JacksonPreset.ts#L20
2. You might have a [property of type `ConstrainedDictionaryModel`](https://github.com/asyncapi/modelina/blob/next/docs/internal-model.md#the-constrained-meta-model) with a serialization type `unwrap` meaning it's not supposed to be interpreted as a normal property, but unwrapped into the enclosing JSON object (if you want JSON encoding of course). Many times you need to explicitly tell the libraries of this i.e. need a preset for it. The reason comes from the `additionalProperties` and `patternProperties` keyword in JSON Schema, where you don't know the property but only its type.

I am not telling you to create this preset, only that it will probably exist in the future, so shutting the door here does not make much sense  🙂 Doing something like this is entirely fine in this PR:

```suggestion
### To and from JSON

Currently not supported, [let everyone know you need it](https://github.com/asyncapi/modelina/issues/new?assignees=&labels=enhancement&template=enhancement.md)!

### To and from XML

Currently not supported, [let everyone know you need it](https://github.com/asyncapi/modelina/issues/new?assignees=&labels=enhancement&template=enhancement.md)!

### To and from binary

Currently not supported, [let everyone know you need it](https://github.com/asyncapi/modelina/issues/new?assignees=&labels=enhancement&template=enhancement.md)!
```